File: app/src/main/java/com/example/purva/ubereats/ui/address/AddressPresenter.java
package com.example.purva.ubereats.ui.address;

import android.content.Context;
import android.content.Intent;

import com.example.purva.ubereats.ui.location.MapsActivity;

public class AddressPresenter implements IAddressPresenter {

    Context context;

    public AddressPresenter(Context context) {
        this.context = context;
    }

    @Override
    public void onClickEnter() {
        Intent intent = new Intent(context, MapsActivity.class);
        context.startActivity(intent);
    }


}

File: app/src/main/java/com/example/purva/ubereats/ui/address/AddressPresenter.kt
package com.example.purva.ubereats.ui.address

import android.content.Context
import android.content.Intent

import com.example.purva.ubereats.ui.location.MapsActivity
import com.example.purva.ubereats.ui.ordersummary.OrderSummary

class AddressPresenter(internal var context: Context) : IAddressPresenter {
    override fun onClickSummary() {
        val intent = Intent(context, OrderSummary::class.java)
        context.startActivity(intent)
    }

    override fun onClickEnter() {
        val intent = Intent(context, MapsActivity::class.java)
        context.startActivity(intent)
    }

}
